### DISCUSSION TITLE
Scaffold with different tolerances at different regions of the model

### DISCUSSION THREAD

**Discussion by dkowollik** [ansys/pyprimemesh#865] [Q&A]
In certain regions my topo faces are  approx. 1mm high, in other regions they are several meters large. Therefore, I have to define different tolerances for the scaffold operation. How can I select these regions and create multiple scaffold operations? In Mechanical GUI I only have to create a Named Selection with a Worksheet, where I can select by coordinate ranges, add a connect object, assign the named selection and define a smaller tolerance. This works just fine for the prime mesher in Mechanical.

**Comment by waltersma** [comment]
Hi @dkowollik,

Thanks for the question.

Are you able to add labels/ named selections in an upstream CAD package to the various entities? You will likely have lots of functionality for identifying and labelling these entities in a geometry application.

This would be the easiest approach.  Could you use pyansys geometry to loop through all the entities and set criteria for labelling them before exporting?

Thanks

**Comment by dkowollik** [comment]
Hi @waltersma,

Unfortunately, the Rhino *.3dm reader still does not support Named Selections I could create in Rhino directly. This would be the best feature to have. I plan to build also optimization workflows in the near future. The problem is the overhead with all the different ansys services and different geometry interpretations involved. Different geometry intrepretations can lead to issues with broken surfaces. We have to use a lot of trimmed nurbs surfaces. Our current Workbench worklow (Rhino->SpaceClaim->Mechanical) sometimes generates edge splits going from SpaceClaim to Mechanical, where we did not have them in SpaceClaim. I would like to keep the workflow as lean as possible. So far you do not have an in memory transfer between different services. Therefore, I would have to 
1. start the GeometryService (Discovery or Spaceclaim >45s, directly GeometryService approx. 18s), 
2. load the *.3dm file (approx. 10s), 
3. create Named Selections (approx. 5s), 
4. write *.fmd file (approx. 5s), 
5. start prime mesh, 
6. read fmd file (> 24s -> this is very long), 
7. start meshing (approx. 8s)
8. write MAPDL cdb file
9. PyACP, ...

Starting GeometryService and importing and exporting files takes a lot of overhead. It would be so much better, to do the labelling workflow and have the search operations to identify topo edges and topo faces directly in PyPrime. Do you have these features or plan to add them?
